File: __aacontest/today/2.py
class Solution:
    def minOperations(self, k: int) -> int:
        operations = 0
        nums = [1]
        
        while sum(nums) < k:
            max_num = max(nums)
            if sum(num * 2 for num in nums) >= k:
                nums[nums.index(max_num)] *= 2
                break
            else:
                nums = [num + 1 for num in nums]
            operations += 1
        
        return operations

# Test cases
solution = Solution()
print(solution.minOperations(11))  # Output: 5
print(solution.minOperations(1))   # Output: 0
print(solution.minOperations(2))   # FIALS WITH 0, EXPECTED: 1

File: __aacontest/today/2.js
var minOperations = function(k) {
  let operations = 0;
  let nums = [1];
  
  while (nums.reduce((acc, val) => acc + val, 0) < k) {
      const maxNum = Math.max(...nums);
      if (nums.reduce((acc, val) => acc + (val * 2), 0) >= k) {
          nums[nums.indexOf(maxNum)] *= 2;
          break;
      } else {
          nums = nums.map(num => num + 1);
      }
      operations++;
  }
  
  return operations;
};

// Test cases
console.log(minOperations(11)); // Output: 5 EXPECTED 5 | PASS
console.log(minOperations(1)); // Output: 0 EXPECTED 0 | PASS
console.log(minOperations(2)); // Output: 0 EXPECTED: 1 | FAIL
